codegen.cpp

```diff
@@ -22,16 +22,28 @@ void NodeAdd::codegen() {
     lhs->codegen();
     cout << "  str x0, [sp, -16]!" << endl;
     rhs->codegen();
+    if(c_type->isPtrType()) {
+        cout << "  mov x1, 8" << endl;
+        cout << "  mul x0, x0, x1" << endl;
+    }
     cout << "  ldr x1, [sp], 16" << endl;
     cout << "  add x0, x0, x1" << endl;
 }
-
 void NodeSub::codegen(){
     lhs->codegen();
     cout << "  str x0, [sp, -16]!" << endl;
     rhs->codegen();
+    if(lhs->c_type->isPtrType() && rhs->c_type->isIntType()) {
+        cout << "  mov x1, 8" << endl;
+        cout << "  mul x0, x0, x1" << endl;
+    }
     cout << "  ldr x1, [sp], 16" << endl;
     cout << "  sub x0, x1, x0" << endl;
+    if(lhs->c_type->isPtrType() && rhs->c_type->isPtrType()) {
```

Comment:
Missing implementation of num - ptr case, if that's supported which I still don't know.